products/views.py

```diff
@@ -1,3 +1,187 @@
+                        'title'     : product.title,
+                        'price'     : product.price,
+                        'region'    : product.region,
+                        'is_group'  : product.is_group,
+                        'bgimg'     : product.background_url,
+                        'userImg'   : product.host.profile_url,
+                        'name'      : product.host.user.name,
+                        'nick'      : product.host.nickname,
+                        'like_count': product.likes.all().count()
+                    } for product in products]
+
+                elif not cache.get('products'):                    
+                    products = Product.objects.select_related('host','host__user').prefetch_related(Prefetch('likes',queryset=Like.objects.all())).all().order_by(ordering)
+
+                    results = [{
+                        'id'        : product.id,
+                        'title'     : product.title,
+                        'price'     : product.price,
+                        'region'    : product.region,
+                        'is_group'  : product.is_group,
+                        'bgimg'     : product.background_url,
+                        'userImg'   : product.host.profile_url,
+                        'name'      : product.host.user.name,
+                        'nick'      : product.host.nickname,
+                        'like_count': product.likes.all().count()
+                    } for product in products]
+
+                    cache.set('products',results)
+                    results = cache.get('products')
+                else: 
+                    results = cache.get('products')
+
+                return JsonResponse({"message":results},status=200)
+
+        except TypeError:
+            return JsonResponse({"message":"TYPE_ERROR"},status=400)
+        except ValueError:
+            return JsonResponse({"message":"VALUE_ERROR"},status=400)
+
+    def post(self, request):
+        try: 
+            data    = json.loads(request.body)
+            # user    = request.user
+            user = User.objects.get(id=1)
+            product = Product.objects.get(id=data['productID'])    
+            Like.objects.update_or_create(user=user, product=product, defaults={'like':data['like']})    
+            return JsonResponse({'message' : 'SUCCESS'}, status=200)
+        
+        except ObjectDoesNotExist:
+            return JsonResponse({'message' : 'MODEL_ERROR'}, status=400)
+        except KeyError:
+            return JsonResponse({'message' : 'KEY_ERROR'}, status=400)
+        except ValueError:
+            return JsonResponse({'message' : 'VALUE_ERROR'}, status=400)
+        except TypeError:
+            return JsonResponse({"message":"TYPE_ERROR"},status=400)
+        except ValidationError:
+            return JsonResponse({"message":"VALIDATION_ERROR"},status=400)
+            
+
+
+
+class ProductDetailView(View): 
+    @query_debugger
```

Comment:
디버깅 데코레이터는 개발할때만 사용하셔야되고, 실제로 merge 및 배포하실때는 제거하셔야 합니다.
이유는 프린트 자체만으로도 실행 속도에 영향을 주기 때문입니다.

```suggestion
    @query_debugger
```

Reply:
병민님이 공유해주신 @query_debugger 데코레이터 ORM최적화에 엄청난 도움이 되었습니다! 감사드려요~~

Reply:
도움이 되었다니 너무너무 다행이에요 ㅎㅎ 저도 감사합니다!!!!